### ISSUE TITLE
Connect to a remote spark cluster (Databricks Connect)

### ISSUE THREAD

**Issue by crackityjones666** [dotnet/spark#516]
Hi!

I need to refactor a python api that uses pyspark to execute a sql statement on Spark cluster in Azure Databricks.  It uses Databricks connect (https://docs.databricks.com/dev-tools/databricks-connect.html) to initialize the connectivity to the cluster.

Is it possible to achieve something like this in c#, I would love to have an mvc core api!

Thanks!

**Comment by imback82** [comment]
We are working with Databricks engineers to be able to use C# with databricks-connect. We will update this thread when we get more info. cc: @elvaliuliuliu

**Comment by crackityjones666** [comment]
Thank you!  My best bet would be to keep this python api for now?  Or is there another way to have my web service trigger a SQL query against a Databricks model?  Also, is there an ETA around this?  

Thank you very much for such a quick reply!

**Comment by shadowmint** [comment]
I have posted this on https://ideas.databricks.com/ideas/

However, is this really up to the databricks folk to implement?

I'm not sure it is... my understanding is that the databricks connect libraries for python, r, etc. rely on a vendored set of .jar files that are shipped when you install databricks connect.

ie. After having performed the basic setup:

```
python3 -m venv env
source ./env/bin/activate
pip install databricks-connect
databricks-connect configure
databricks-connect test
```

Then you can find the location of the installed 'modified' spark implementation using:

```
(env) M:tmp doug$ databricks-connect get-spark-home
/Users/doug/dev/wok/tmp/env/lib/python3.5/site-packages/pyspark
(env) M:tmp doug$ databricks-connect get-jar-dir
/Users/doug/dev/wok/tmp/env/lib/python3.5/site-packages/pyspark/jars
```

However, it seems currently what is lacking is an implementation in Microsoft.Spark to allow those values to be used as overrides in the library? Specifically the jars folder.

Am I mistaken in my understanding?

I've seen multiple threads saying this is something that needs to be implemented by the Databricks folk, but my understanding is that, that is not correct. 

The databricks connect implementation for arbitrary consumers *already exists*, it is simply:

1) Not supported in Microsoft.Spark

2) Not documented how to set it up on https://docs.databricks.com/dev-tools/databricks-connect.html

tldr; Seems like game of hand ball here to me, and I'm not sure it's fair to say this is being blocked by the Databricks side...

**Comment by imback82** [comment]
@shadowmint, .NET for Apache Spark works with databricks connect as long as you don't use UDFs written in C#. 

Databricks connect works by sending the logical plans to the server side and while the plans are parsed on the server side, the `PythonFunction.pythonExec` value (which is the .NET Spark Worker) passed was not being respected (overwritten as a Python executable).

I had a call with a Databricks engineer and got a confirmation that the library needs to be updated. I will update this thread with the latest status.

Thanks.
